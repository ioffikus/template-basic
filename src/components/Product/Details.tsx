import React from 'react';
import { connect } from 'react-redux';
import { Tabs, Table } from 'antd';
import { Link } from 'react-router-dom';
import _ from 'lodash';
import Price from 'src/components/Price';

interface IOwnProps {}

interface IConnectedState {
  ui: Alicanto.Models.UI;
  uid: string;
  schemaFields: Alicanto.Models.CatalogItemSchemaFields;
  categories: Alicanto.Models.Category[];
  modifications: Alicanto.Models.CatalogItem[];
  isBundle: boolean;
  bundleContent: Alicanto.Models.CatalogItem[];
}

type IProps = IOwnProps & IConnectedState;

interface IState {
  activeKey: string;
  specificationsData: { key: string; label: string; value: string }[];
}

const mapStateToProps = (state: Store.IState, ownProps: IOwnProps) => ({
  ui: state.ui,
  uid: state.product.uid,
  schemaFields: state.product.schema_fields,
  categories: state.product.categories, // TODO: design is not released yet
  modifications: state.product.modifications, // TODO: design is not released yet
  isBundle: state.product.is_bundle, // TODO: design is not released yet
  bundleContent: state.product.bundle_content, // TODO: design is not released yet
});

const WrappedDetails = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Details extends React.Component<IProps, IState> {
    readonly defaultKey = 'specifications';

    readonly specificationsColumns = [{ dataIndex: 'label' }, { dataIndex: 'value' }];

    state: IState = {
      specificationsData: this.getSpecificationsData(this.props.schemaFields),
      activeKey: this.defaultKey,
    };

    componentWillReceiveProps(nextProps: IProps) {
      if (this.props.uid !== nextProps.uid) {
        const specificationsData = this.getSpecificationsData(nextProps.schemaFields);
        this.setState({ specificationsData, activeKey: this.defaultKey });
      }
    }

    handleChange = (activeKey: string) => {
      this.setState({ activeKey });
    };

    getSpecificationsData(schemaFields: Alicanto.Models.CatalogItemSchemaFields) {
      const result: { key: string; label: string; value: string }[] = [];
      Object.keys(schemaFields).forEach(key => {
        if (key === 'picture_url') {
          return;
        }
        const schemaField = _.get(schemaFields, key);
        const inList = _.get(schemaField, 'in_list', null);
        if (!_.isNull(inList)) {
          const label = _.get(schemaField, 'label', '');
          const value = _.get(schemaField, 'value', '');
          result.push({ key, label, value });
        }
      });
      return result;
    }

    render() {
      return (
        <div className="root">
          <style jsx>{`
            .root :global(.ant-tabs-bar) {
              text-transform: uppercase;
            }
            .root :global(.ant-tabs-tabpane) {
              padding-left: 26px;
              padding-right: 26px;
            }
            .root :global(.ant-table) {
              font-size: 14px;
            }
            .root :global(.ant-table-tbody > tr > td) {
              padding-top: 17px;
              padding-bottom: 17px;
            }
            .root :global(.ant-table td:first-of-type) {
              color: #b2bdcb;
            }
            .root :global(.ant-table tr:hover > td) {
              background: transparent;
            }
            li {
              padding-top: 17px;
              padding-bottom: 17px;
            }
            .root :global(.bundle),
            .root :global(.modification),
            .root :global(.category) {
              font-size: 14px;
            }
            .root :global(.modification .currency) {
              font-size: inherit;
              color: inherit;
            }
          `}</style>
          <Tabs activeKey={this.state.activeKey} onChange={this.handleChange}>
            <Tabs.TabPane tab="Specifications" key="specifications">
              <Table
                columns={this.specificationsColumns}
                dataSource={this.state.specificationsData}
                pagination={false}
                showHeader={false}
              />
            </Tabs.TabPane>
            {!_.isEmpty(this.props.bundleContent) && (
              <Tabs.TabPane tab="Bundle" key="bundle">
                <ul>
                  {this.props.bundleContent.map(bundle => {
                    const schemaFields = bundle.schema_fields;
                    const shortDescription = _.get(schemaFields, 'short_description.value', '');
                    const name = _.get(schemaFields, 'name.value', '');
                    return (
                      <li key={bundle.uid}>
                        <Link to={`/product/${bundle.uid}`} title={shortDescription} className="bundle">
                          {name}
                        </Link>
                      </li>
                    );
                  })}
                </ul>
              </Tabs.TabPane>
            )}
            {!_.isEmpty(this.props.modifications) && (
              <Tabs.TabPane tab="Modifications" key="modifications">
                <ul>
                  {this.props.modifications.map(modification => {
                    const schemaFields = modification.schema_fields;
                    const shortDescription = _.get(schemaFields, 'short_description.value', '');
                    const name = _.get(schemaFields, 'name.value', '');
                    const currency: string = _.get(schemaFields, 'currency.value', '');
                    const price = _.get(schemaFields, 'price.value', 0);
                    return (
                      <li key={modification.uid}>
                        <Link to={`/product/${modification.uid}`} title={shortDescription} className="modification">
                          {name} -&nbsp;
                          <Price price={price} currency={currency} />
                        </Link>
                      </li>
                    );
                  })}
                </ul>
              </Tabs.TabPane>
            )}
            {!_.isEmpty(this.props.categories) && (
              <Tabs.TabPane tab="Categories" key="categories">
                <ul>
                  {this.props.categories.map(category => {
                    return (
                      <li key={category.uid}>
                        <Link to={`/catalog/${category.uid}`} className="category">
                          {category.name}
                        </Link>
                      </li>
                    );
                  })}
                </ul>
              </Tabs.TabPane>
            )}
          </Tabs>
        </div>
      );
    }
  },
);

export default WrappedDetails;
