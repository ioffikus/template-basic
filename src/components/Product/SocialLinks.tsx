import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import cx from 'classnames';
import Svg from 'src/components/Svg';

interface IOwnProps {}

interface IConnectedState {
  ui: Alicanto.Models.UI;
}

type IProps = IOwnProps & IConnectedState;

interface IState {}

const mapStateToProps = (state: Store.IState, ownProps: IOwnProps) => ({
  ui: state.ui,
});

const WrappedSocial = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Social extends React.Component<IProps, IState> {
    render() {
      const itemsUI = this.props.ui.routes.productCard.socialLinks.items;
      const visibleItems = _.transform(
        itemsUI,
        (result, value, key) => {
          const isVisible = value.settings.isVisible;
          if (isVisible) {
            result.push({ ...value, name: key });
          }
        },
        [],
      );
      return (
        <div className="root">
          <style jsx>{`
            .list {
              display: flex;
              align-items: flex-end;
              justify-content: flex-end;
              margin: 0;
              padding: 0;
              list-style-type: none;
            }
            .root :global(.link) {
              font-size: 0;
            }
            .item {
              display: flex;
              align-items: center;
              justify-content: center;
              width: 25px;
              height: 25px;
            }
            .root :global(.icon-facebook) {
              fill: #008fff;
              width: 12px;
              height: 17px;
            }
            .root :global(.icon-twitter) {
              fill: #009cff;
              width: 11px;
              height: 17px;
            }
            .root :global(.icon-vkontakte) {
              fill: #4c75a3;
              width: 15px;
              height: 12px;
            }
            .root :global(.icon-instagram) {
              fill: #ff0070;
              width: 14px;
              height: 14px;
            }
          `}</style>
          <ul className="list">
            {visibleItems.map(item => {
              const link = item.i18n.defaultText;
              const name = item.name;
              return (
                <li className="item" key={name}>
                  <a href={link} target="_blank" rel="noopener noreferrer" className="link">
                    <Svg
                      name={name}
                      className={cx('icon', {
                        [`icon-${name}`]: name,
                      })}
                    />
                    {name}
                  </a>
                </li>
              );
            })}
          </ul>
        </div>
      );
    }
  },
);

export default WrappedSocial;
