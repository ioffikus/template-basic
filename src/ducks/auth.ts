import { createAction, createReducer } from 'redux-act';
import { AxiosResponse } from 'axios';
import { push } from 'connected-react-router';
import { GET, POST } from 'src/core/API';
import EndpointsConfig from 'src/configs/endpoints';

export const REDUCER = 'AUTH';
const NS = `${REDUCER}__`;

export interface IAuthState extends Store.IReduxCommonState {
  isAuthenticated: boolean;
  isAuthLinkRequesting: boolean;
  url: string;
  data: object;
}

export const initialState: IAuthState = {
  isAuthenticated: false,
  isRequesting: false,
  isRequestError: false,
  isAuthLinkRequesting: false,
  url: null,
  data: null,
};

const reducer = createReducer<IAuthState>({}, initialState);

export const loginUserSuccess = createAction(`${NS}LOGIN_USER_SUCCESS`);
reducer.on(loginUserSuccess, state => ({
  ...state,
  isRequesting: false,
  isRequestError: false,
  isAuthenticated: true,
}));

const authLinkSuccess = createAction<{ url: string; data: object }>(`${NS}AUTH_LINK_SUCCESS`);
reducer.on(authLinkSuccess, (state, { url, data }) => ({
  ...state,
  url,
  data: { ...data },
  isAuthLinkRequesting: false,
}));

const loginUserFailure = createAction(`${NS}LOGIN_USER_FAILURE`);
reducer.on(loginUserFailure, state => ({
  ...state,
  isRequesting: false,
  isAuthenticated: false,
  isRequestError: true,
}));

const authLinkFailure = createAction(`${NS}AUTH_LINK_FAILURE`);
reducer.on(authLinkFailure, state => ({
  ...state,
  isAuthLinkRequesting: false,
  url: null,
  data: null,
}));

const loginUserRequest = createAction(`${NS}LOGIN_USER_REQUEST`);
reducer.on(loginUserRequest, state => ({
  ...state,
  isRequesting: true,
}));

const authLinkRequest = createAction(`${NS}AUTH_LINK_REQUEST`);
reducer.on(authLinkRequest, state => ({
  ...state,
  isAuthLinkRequesting: true,
  url: null,
  data: null,
}));

const reset = createAction(`${NS}RESET`);
reducer.on(reset, () => ({
  ...initialState,
}));

export const getAuthLink = (): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(authLinkRequest());
    GET(
      {
        url: `${EndpointsConfig.AUTH.AUTH_LINK}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        const { url, data } = resp.data as { url: string; data: object };
        // WIP does not use Alicanto.API.GET_AUTH_OAUTH_PROVIDER_URL_RESPONSE for develop
        dispatch(authLinkSuccess({ url, data }));
        resolve();
      })
      .catch(error => {
        dispatch(authLinkFailure());
        reject(error);
      });
  });
};

export const loginUser = (username: string, password: string): Store.IReduxThunkCb => (
  dispatch,
  getState,
  xShopHeaders,
) => {
  return new Promise((resolve, reject) => {
    dispatch(loginUserRequest());
    const data: Alicanto.API.POST_AUTH_LOGIN_BODY = {
      username,
      password,
    };
    POST(
      {
        data,
        url: `${EndpointsConfig.AUTH.LOGIN}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(loginUserSuccess());
        if (process.env.BROWSER) {
          // TODO: move to external
          // redirect to home page
          dispatch(push('/'));
        }
        resolve();
      })
      .catch(error => {
        dispatch(loginUserFailure());
        reject(error);
      });
  });
};

export const logoutUser = (cb: () => void): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(loginUserRequest());
    GET(
      {
        url: `${EndpointsConfig.AUTH.LOGOUT}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(loginUserSuccess());
        dispatch(reset());
        if (cb) {
          cb();
        }
        if (process.env.BROWSER) {
          // TODO: move to external
          // redirect to home page
          dispatch(push('/'));
        }
        resolve();
      })
      .catch(error => {
        dispatch(loginUserFailure());
        reject(error);
      });
  });
};

export default reducer;
