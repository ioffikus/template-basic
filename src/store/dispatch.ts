import { Store } from 'redux';
import _ from 'lodash';
import { showError } from 'src/store/common';

let store: Store<Store.IState>;

export const setupDispatch = (config: { store: Store<Store.IState> }) => {
  store = config.store;
};

const dispatch = (action: any) => {
  if (_.isFunction(action)) {
    return store.dispatch(action).catch((error: any) => {
      if (!_.isNull(error)) {
        showError(error);
      }
      return Promise.reject(error);
    });
  }
  return store.dispatch(action);
};

export default dispatch;
