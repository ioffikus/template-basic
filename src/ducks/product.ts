import { createAction, createReducer } from 'redux-act';
import { AxiosResponse } from 'axios';
import { matchPath } from 'react-router';
import _ from 'lodash';
import { GET } from 'src/core/API';
import EndpointsConfig from 'src/configs/endpoints';
import { setMeta } from 'src/ducks/meta';

export const REDUCER = 'PRODUCTS';
const NS = `${REDUCER}__`;

export interface IProductState extends Alicanto.API.GET_CATALOG_UID_RESPONSE, Store.IReduxCommonState {}

export const initialState: IProductState = {
  uid: null,
  id: null,
  is_unlimited: null,
  schema_fields: {},
  categories: null,
  modifications: null,
  pictures: null,
  isRequesting: false,
  isRequestError: false,
};

const reducer = createReducer<IProductState>({}, initialState);

const callSuccess = createAction<Alicanto.API.GET_CATALOG_UID_RESPONSE>(`${NS}CALL_SUCCESS`);
reducer.on(callSuccess, (state, { id, uid, schema_fields, categories, modifications, pictures, is_unlimited }) => ({
  ...state,
  uid,
  id: _.isNil(id) ? null : id,
  is_unlimited: _.isNil(is_unlimited) ? null : is_unlimited,
  schema_fields: { ...schema_fields },
  categories: _.isNil(categories) ? null : [...categories],
  modifications: _.isNil(modifications) ? null : [...modifications],
  pictures: _.isNil(pictures) ? null : [...pictures],
  isRequesting: false,
  isRequestError: false,
}));

const callFailure = createAction(`${NS}CALL_FAILURE`);
reducer.on(callFailure, state => ({
  ...state,
  uid: null,
  id: null,
  is_unlimited: null,
  schema_fields: {},
  categories: null,
  modifications: null,
  pictures: null,
  isRequestError: true,
}));

const callRequest = createAction(`${NS}CALL_REQUEST`);
reducer.on(callRequest, state => ({
  ...state,
  isRequesting: true,
}));

export const getProduct = (abortToken: string): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(callRequest());
    const router = getState().router;
    const match = matchPath(router.location.pathname, { path: '/product/:uid' });
    const uid = _.get(match, 'params.uid', null);
    GET(
      {
        url: `${EndpointsConfig.PRODUCT}${uid}`,
      },
      xShopHeaders,
      abortToken,
    )
      .then((resp: AxiosResponse) => {
        const data = resp.data as Alicanto.API.GET_CATALOG_UID_RESPONSE;
        dispatch(callSuccess(data));
        resolve();
      })
      .catch(error => {
        dispatch(callFailure());
        reject(error);
      });
  }).then(() => {
    const schemaFields = getState().product.schema_fields;
    const name = _.get(schemaFields, 'name.value', '');
    const description = _.get(schemaFields, 'description.value', '');
    dispatch(setMeta({ description, title: name, keywords: `${name} ${description}` }));
  });
};

export default reducer;
