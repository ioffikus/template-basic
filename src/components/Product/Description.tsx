import React from 'react';
import { connect } from 'react-redux';
import { Button, Modal } from 'antd';
import _ from 'lodash';
import dispatch from 'src/store/dispatch';
import { IProductState } from 'src/ducks/product';
import { addToCart } from 'src/ducks/cart';
import CartItems from 'src/components/CartItems';
import Price from 'src/components/Price';
import GoToCart from 'src/components/GoToCart';

interface IOwnProps {}

interface IConnectedState {
  ui: Alicanto.Models.UI;
  product: IProductState;
}

type IProps = IOwnProps & IConnectedState;

interface IState {
  isVisibleCartModal: boolean;
}

const mapStateToProps = (state: Store.IState, ownProps: IOwnProps) => ({
  ui: state.ui,
  product: state.product,
});

const WrappedDescription = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Description extends React.Component<IProps, IState> {
    state: IState = {
      isVisibleCartModal: false,
    };

    handleAddToCart = () => {
      dispatch(
        addToCart(this.props.product, () => {
          this.setState({
            isVisibleCartModal: true,
          });
        }),
      );
    };

    handleCartModalCancel = () => {
      this.setState({
        isVisibleCartModal: false,
      });
    };

    render() {
      const schemaFields = this.props.product.schema_fields;
      // const isNewItem = false; // TODO: API & design is not released yet [ch1801]
      // const isSale = false; // TODO: API & design is not released yet [ch1801]
      const name = _.get(schemaFields, 'name.value', '');
      const description = _.get(schemaFields, 'description.value', '');
      const currency: string = _.get(schemaFields, 'currency.value', '');
      const price = _.get(schemaFields, 'price.value', 0);
      // const oldPrice = 0; // TODO: API & design is not released yet [ch1801]
      // const isShowTags = isNewItem || isSale; [ch1803]
      const productCardUI = this.props.ui.routes.productCard;
      const commonUI = this.props.ui.common;
      return (
        <div className="root">
          <style jsx>{`
            .root {
              font-size: 14px;
            }
            .name {
              font-size: 24px;
              line-height: 1.17;
              margin-bottom: 15px;
              color: inherit;
            }
            .description {
              line-height: 1.7;
            }
            .root :global(.text + .text) {
              margin-top: 15px;
            }
            .content {
              margin-bottom: 17px;
            }
            .buttons-wrapper {
              display: flex;
              justify-content: space-between;
            }
            .root :global(.select-size) {
              // width: 100%;
              font-size: inherit;
              color: #a7b4c3;
              margin-right: 10px;
            }
            .root :global(.ant-select-lg .ant-select-selection--single) {
              height: 40px;
              background-color: #e5ecf4;
              border-radius: 8px;
              border: 0;
            }
            .root :global(.ant-select-lg .ant-select-selection__rendered) {
              line-height: 40px;
              margin-left: 25px;
              margin-right: 15px;
            }
            .root :global(.button-price) {
              height: 40px;
              font-weight: 500;
              letter-spacing: 2px;
              border-radius: 8px;
              padding: 0 35px;
            }
            .root :global(.button-price .currency) {
              color: inherit;
              font-size: inherit;
            }
          `}</style>
          <div className="content">
            <h1 className="name">{name}</h1>
            <p className="description">{description}</p>
          </div>
          <div className="buttons-wrapper">
            {/* TODO [ch1167] */}
            {/* <Select defaultValue="Size" size="large" className="select-size">
              <Select.Option value="5.5">5.5</Select.Option>
              <Select.Option value="6">6</Select.Option>
              <Select.Option value="6.5">6.5</Select.Option>
              <Select.Option value="7">7</Select.Option>
              <Select.Option value="7.5">7.5</Select.Option>
              <Select.Option value="8">8</Select.Option>
              <Select.Option value="8.5">8.5</Select.Option>
              <Select.Option value="9">9</Select.Option>
              <Select.Option value="9.5">9.5</Select.Option>
              <Select.Option value="10">10</Select.Option>
              <Select.Option value="10.5">10.5</Select.Option>
              <Select.Option value="11">11</Select.Option>
            </Select> */}
            <Button type="primary" size="large" className="button-price" onClick={this.handleAddToCart}>
              {productCardUI.button.i18n.defaultText} &nbsp;
              <Price price={price} currency={currency} />
            </Button>
          </div>
          <Modal
            visible={this.state.isVisibleCartModal}
            onCancel={this.handleCartModalCancel}
            footer={false}
            width={540}
          >
            <CartItems
              mayChangeAmountItems={true}
              footer={[
                <Button
                  key="continueShopping"
                  className="button"
                  type="primary"
                  size="large"
                  onClick={this.handleCartModalCancel}
                >
                  {commonUI.buttons.continueShopping.i18n.defaultText}
                </Button>,
                <GoToCart key="goToCart">{commonUI.buttons.goToCart.i18n.defaultText}</GoToCart>,
              ]}
            />
          </Modal>
        </div>
      );
    }
  },
);

export default WrappedDescription;
