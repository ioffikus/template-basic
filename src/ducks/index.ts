// instead this, we have router in store via connected-react-router
// import { routerReducer as routing, RouterState } from 'react-router-redux';
import { combineReducers } from 'redux';
import auth from './auth';
import cart from './cart';
import cartPaidItems from './cartPaidItems';
import catalogQuery from './catalogQuery';
import catalog from './catalog';
import categories from './categories';
import categoriesMenu from './categoriesMenu';
import filter from './filter';
import locale from './locale';
import meta from './meta';
import page from './page';
import product from './product';
import progress from './progress';
import popularItems from './popularItems';
import recommendItems from './recommendItems';
import routes from './routes';
import ui from './ui';
import user from './user';
import userDelivery from './userDelivery';
import userOrders from './userOrders';

export default combineReducers<Store.IState>({
  ui,
  auth,
  user,
  categories,
  categoriesMenu,
  catalogQuery,
  catalog,
  product,
  progress,
  routes,
  meta,
  filter,
  cart,
  cartPaidItems,
  userDelivery,
  userOrders,
  popularItems,
  recommendItems,
  page,
  locale,
});
