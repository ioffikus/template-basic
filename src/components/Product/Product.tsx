import React from 'react';
import { Row, Col } from 'antd';
import dispatch from 'src/store/dispatch';
import { getProduct } from 'src/ducks/product';
import Breadcrumbs from 'src/components/Breadcrumbs';
import ImagePreview from 'src/components/ProductImagePreview';
import SocialLinks from './SocialLinks';
import Description from './Description';
import Details from './Details';

interface IProps {}

interface IState {}

class Product extends React.Component<IProps, IState> {
  static fetchData: Store.IFetchDataFn = abortToken => {
    return dispatch(getProduct(abortToken));
  };

  render() {
    return (
      <div className="root app-container">
        <style jsx>{`
          .root :global(.header) {
            margin-bottom: 55px;
            padding-bottom: 4px;
            border-bottom: 1px solid #e5ecf4;
          }
          .root :global(.content) {
            margin-bottom: 20px;
          }
        `}</style>
        <Row type="flex" justify="space-between" align="bottom" className="header">
          <Col span={20}>
            <Breadcrumbs for="product" />
          </Col>
          <Col span={4}>
            <SocialLinks />
          </Col>
        </Row>
        <Row type="flex" gutter={30}>
          <Col xs={24} sm={24} md={12} className="content">
            <ImagePreview />
          </Col>
          <Col xs={24} sm={24} md={12} className="content">
            <Description />
          </Col>
        </Row>
        <Details />
      </div>
    );
  }
}

export default Product;
