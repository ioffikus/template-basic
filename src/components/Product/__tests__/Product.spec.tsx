import React from 'react';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import nanoid from 'nanoid';
import { setupDispatch } from 'src/store/dispatch';
import { getProduct } from 'src/ducks/product';
import Product from '../Product';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
jest.mock('src/ducks/product', () => {
  return {
    getProduct: jest.fn(() => {
      return () => Promise.resolve();
    }),
  };
});

describe(null, () => {
  it('fetchData()', () => {
    const store = mockStore({});
    setupDispatch({ store });
    const abortToken = nanoid();
    return Product.fetchData(abortToken).then(() => {
      expect(getProduct).toBeCalledWith(abortToken);
    });
  });
  it('render()', () => {
    const store = mockStore({});
    setupDispatch({ store });
    const wrapper = shallow(<Product />, { context: { store } });
    expect(wrapper).toMatchSnapshot();
  });
});
