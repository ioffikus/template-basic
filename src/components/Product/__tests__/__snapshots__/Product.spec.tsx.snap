// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() 1`] = `
<div
  className="jsx-3486026754 root app-container"
>
  <JSXStyle
    css=".root.jsx-3486026754 .header{margin-bottom:55px;padding-bottom:4px;border-bottom:1px solid #e5ecf4;}.root.jsx-3486026754 .content{margin-bottom:20px;}"
    styleId="3486026754"
  />
  <Row
    align="bottom"
    className="header"
    gutter={0}
    justify="space-between"
    type="flex"
  >
    <Col
      span={20}
    >
      <Connect(Breadcrumbs)
        for="product"
      />
    </Col>
    <Col
      span={4}
    >
      <Connect(Social) />
    </Col>
  </Row>
  <Row
    gutter={30}
    type="flex"
  >
    <Col
      className="content"
      md={12}
      sm={24}
      xs={24}
    >
      <Connect(ProductImagePreview) />
    </Col>
    <Col
      className="content"
      md={12}
      sm={24}
      xs={24}
    >
      <Connect(Description) />
    </Col>
  </Row>
  <Connect(Details) />
</div>
`;
